date.hpp

```diff
@@ -157,6 +159,32 @@ bool Date::is_last_day_of_year()
     return false;
 }
 
+bool Date::is_last_week_of_month()
```

Comment:
This is not going to work, this implementation expects 4 weeks in a month. It needs to test for certain interval of days.

Reply:
Good catch it should have been >= not ==.